axum-extra/src/extract/host.rs

```diff
@@ -31,31 +35,50 @@ where
     type Rejection = HostRejection;
 
     async fn from_request_parts(parts: &mut Parts, _state: &S) -> Result<Self, Self::Rejection> {
+        parts
+            .extract::<Option<Host>>()
+            .await
+            .ok()
+            .flatten()
+            .ok_or(HostRejection::FailedToResolveHost(FailedToResolveHost))
```

Comment:
A nitpick: the `OptionalFromRequestParts` implementation contains less information than the `FromRequestParts` one. In the future, we could, for example, return other errors than just `HostRejection::FailedToResolveHost`.
I'd suggest to keep the implementation of `FromRequestParts` and to base the `OptionalFromRequestParts` implementation on top of it, removing the error information.

Reply:
@yanns if we actually introduce other rejections for `Host` then we could change the `OptionalFromRequestParts` implementation to reject with `HostRejection` too. I don't see why we would need separate implementations for this 🤔

Reply:
`OptionalFromRequestParts` returns `Result<Option<T>, Rejection>` 😉

Reply:
ah right, then forget about my comment. I had the feeling that `OptionalFromRequestParts` is swallowing errors and was less precise than `FromRequestParts`. But I'm probably wrong here.